[NUI] Add Outlined and TextOnly Button styles

jaehyun0cho opened the pull request:
### Description of Change ###
<!-- Describe your changes here. -->

The latest Tizen 7.0 UX introduces Outlined and TextOnly style Buttons.
Outlined style Button has transparent background and colored borderline.
TextOnly style Button has a text and transparent background.

### API Changes ###
<!-- If you have the ACR for changing APIs, put the link of the ACR: -->
 - ACR:

<!--
If you can't make the ACR, List all API changes here (or just put None), example:
Added:
 - SafeBundleHandle Bundle.SafeBundleHandle { get; } // Property
 - void Bundle.AddItem(string key, string value);

Changed:
 - object Bundle.ReceiveItem(string key) => object Bundle.GetItem(string key);
-->

jaehyun0cho commented:
This PR should be merged with Rebase and merge.

dongsug-song commented:
@jaehyun0cho 
thank you for the patch. could you explain why these components are needed?
could you update it in description?
thank you.

jaehyun0cho commented:
> @jaehyun0cho thank you for the patch. could you explain why these components are needed? could you update it in description? thank you.

Thank you for your comment.
Tizen 7.0 UX introduces OutlinedButton and TextButton.

everLEEst reviewed (changes requested):
We don't need OutlineButton, TextButton class for this.
adding style in the default theme is enough.

jaehyun0cho commented:
> We don't need OutlineButton, TextButton class for this. adding styles in the default theme are enough.

Thank you for your comment.
I have removed OutlinedButton and TextButton classes.
Instead, I have modified the samples to use "Tizen.NUI.Components.OutlinedButton" and "Tizen.NUI.Components.TextButton" styles.

everLEEst reviewed (approved):
Overall looks good to me.
but before pushing this patch, I think we need to decide naming policy of styles.
I think we have two good solutions.
```c#
Tizen.NUI.Components.Button
Tizen.NUI.Components.OutlinedButton
Tizen.NUI.Components.TextButton
```
or
```
Tizen.NUI.Components.Button  (.Default)
Tizen.NUI.Components.Button.Outlined
Tizen.NUI.Components.Button.TextOnly //Text might quite confusing in this case.
```

jaehyun0cho commented:
> Overall looks good to me. but before pushing this patch, I think we need to decide naming policy of styles. I think we have two good solutions.
> 
> ```cs
> Tizen.NUI.Components.Button
> Tizen.NUI.Components.OutlinedButton
> Tizen.NUI.Components.TextButton
> ```
> 
> or
> 
> ```
> Tizen.NUI.Components.Button  (.Default)
> Tizen.NUI.Components.Button.Outlined
> Tizen.NUI.Components.Button.TextOnly //Text might quite confusing in this case.
> ```

Good point!
I prefer the second way.
Reviewers, please leave your opinions~

jaehyun0cho commented:
I have updated this PR to use the style naming rule as follows.
"Tizen.NUI.Components.Button.Outlined"
"Tizen.NUI.Components.Button.TextOnly"

dongsug-song reviewed (approved):
very good idea~
we can avoid to create new classes which seem to be heavy and rich.
